src/Parser/TypeParser.php

```diff
@@ -858,6 +859,19 @@ private function parseArrayShape(TokenIterator $tokens, Ast\Type\TypeNode $type,
 
 			if ($tokens->tryConsumeTokenType(Lexer::TOKEN_VARIADIC)) {
 				$sealed = false;
+				$tokens->tryConsumeTokenType(Lexer::TOKEN_PHPDOC_EOL);
+				if ($tokens->isCurrentTokenType(Lexer::TOKEN_OPEN_ANGLE_BRACKET)) {
+					$extraItemType = $this->parseGeneric(
+						$tokens,
+						$this->enrichWithAttributes(
+							$tokens,
+							new Ast\Type\IdentifierTypeNode($kind),
```

Comment:
This and the necessary printing adjustments for it are very hacky. I don't like it at all and would rather add support for a `GenericNode` (matching the current grammar for `Generic` which doesn't have a type identifier)
But I'd prefer a second opinion before doing that.

So these are our options:
- Keep it like this and identify the type as `array` / `list`.
  The main benefit of this approach is that the type resolution in phpstan-src would require very little adjustments for this part, because it's exactly built for this. (https://github.com/phpstan/phpstan-src/blob/04b840344a5f403e9a48120d7b4dccb278cab2d2/src/PhpDoc/TypeNodeResolver.php#L640-L669)
  But I don't mind doing more work there at all.
  
  The disadvantage: Semantically & grammatically it's not accurate and it requires workarounds for printing & string conversions.
- Use an empty string as the type identity
  The main benefit is that it works quite well in this repository and no printing  & string conversions workarounds are necessary.
  
  The disadvantage: Semantically & grammatically it still feels inaccurate
- Add a new `GenericNode` (or similar), basically just `GenericTypeNode` without the base type.
  This would be the most accurate because there really is no base type.

  The disdvantage: Ideally this would then be integrated into `GenericTypeNode`, but for backward compatibility reasons that won't be as simple / currently not possible, slightly raising the complexity when the two interact.